src/Microsoft.AspNet.Mvc.Razor/VirtualPathRazorPageFactory.cs

```diff
@@ -67,7 +69,9 @@ public IRazorPage CreateInstance([NotNull] string relativePath)
                     relativeFileInfo,
                     RazorCompilationService.Compile);
 
-                var page = (IRazorPage)_activator.CreateInstance(_serviceProvider, result.CompiledType);
+                var razorPageFactory = _razorPageCache.GetOrAdd(result.CompiledType, CreateFactory);
+
+                var page = (IRazorPage)razorPageFactory(_serviceProvider, null);
```

Comment:
@yishaigalatzer, do views need type activation for views (i.e. do they need services)? Could we go down the route of using `Activator.CreateInstance` or caching the ctor for the compiled type ?